xcdat/regridder/xgcm.py

```diff
@@ -164,17 +164,20 @@ def vertical(self, data_var: str, ds: xr.Dataset) -> xr.Dataset:
 
         target_data: str | xr.DataArray | None = None
 
-        try:
-            target_data = ds[self._target_data]
-        except ValueError:
-            target_data = self._target_data
-        except KeyError as e:
-            if self._target_data is not None and isinstance(self._target_data, str):
-                raise RuntimeError(
-                    f"Could not find target variable {self._target_data!r} in dataset"
-                ) from e
-
-            target_data = None
+        if isinstance(self._target_data, str) and self._target_data == "infer":
+            target_data = self._infer_target_data(ds)
+        else:
+            try:
+                target_data = ds[self._target_data]
+            except ValueError:
+                target_data = self._target_data
+            except KeyError as e:
+                if self._target_data is not None and isinstance(self._target_data, str):
+                    raise RuntimeError(
+                        f"Could not find target variable {self._target_data!r} in dataset"
+                    ) from e
+
+                target_data = None
 
         # assumes new verical coordinate has been calculated and stored as `pressure`
         # TODO: auto calculate pressure reference http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#parametric-v-coord
```

Comment:
Are these comments still relevant? Or does this PR handle the TODO?

Reply:
Not relevant, I'll remove.